pkg/thread/periodic_tasks.go

```diff
@@ -0,0 +1,70 @@
+package thread
+
+import (
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+type PeriodicTask struct {
+	Name     string
+	Interval time.Duration
+	Exec     func()
+}
+
+type PeriodicTasks struct {
+	log   logrus.FieldLogger
+	done  chan struct{}
+	tasks []PeriodicTask
+}
+
+func NewPeriodicTasksRunner(log logrus.FieldLogger, monitors []PeriodicTask) *PeriodicTasks {
+	return &PeriodicTasks{
+		log:   log,
+		tasks: monitors,
+	}
+}
+
+// Start monitors thread
+func (m *PeriodicTasks) Start() {
+	m.done = make(chan struct{})
+	m.log.Infof("Started monitors thread")
+	go m.runTasks()
+}
+
+// Stop monitors
+func (m *PeriodicTasks) Stop() {
+	m.log.Infof("Stopping all monitors")
+	close(m.done)
+	m.log.Infof("Stopped all monitors")
+}
+
+func (m *PeriodicTasks) runTasks() {
+	var wg sync.WaitGroup
+	wg.Add(len(m.tasks))
+	for _, task := range m.tasks {
+		go m.taskRunnerLoop(m.done, &wg, task)
+	}
+	m.log.Infof("Waiting for all tasks to finish")
+	wg.Wait()
+	m.log.Infof("Finished running all tasks")
+}
+
+func (m *PeriodicTasks) taskRunnerLoop(done <-chan struct{}, wg *sync.WaitGroup, task PeriodicTask) {
+	m.log.Infof("Start running %s task", task.Name)
+	defer wg.Done()
+	intervalTimer := time.NewTimer(0)
```

Comment:
As you suggested maybe switch to https://golang.org/pkg/time/#NewTicker
if the interval is long we don't want to wait more then the interval time for the next iteration, 
you can fix thread package as well